### PR TITLE
[7.x] PLANNER-2171 avoid reflection warnings + fix mvel native + upgrade to quarkus 1.8.2

### PR DESCRIPTION
**Pull request by ge0ffrey**
quarkus:dev is still broken, but I need to get this part merged in order to isolate the remaining issue easily and replicate with the quarkus team.

### CODE REVIEW

**Review by triceo — APPROVED** [review]
Don't see why not. When green.

**Comment by ge0ffrey** [comment]
```
Error:  The builder requested using id = smart cannot be found -> [Help 1]
  org.apache.maven.lifecycle.internal.builder.BuilderNotFoundException: The builder requested using id = smart cannot be found
      at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:117)
```

**Comment by ge0ffrey** [comment]
The OS x ones

` Error:  Failed to execute goal on project optaplanner-core: Could not resolve dependencies for project org.optaplanner:optaplanner-core:bundle:7.45.0-SNAPSHOT: Could not transfer artifact com.github.javaparser:javaparser-core:jar:3.13.10 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/com/github/javaparser/javaparser-core/3.13.10/javaparser-core-3.13.10.jar: Connection reset -> [Help 1]`

**Comment by triceo** [comment]
Yeah, a lot of the CI improvements (such as the Maven cache) have not been back-ported to 7.x.

**Review by rsynek — COMMENTED** [review]
Would you please upgrade the Quarkus also in the build.gradle over here:
https://github.com/kiegroup/optaplanner/blob/7.x/optaplanner-quickstarts/quarkus-school-timetabling/build.gradle ?

**Comment by rsynek** [comment]
Ad CI failures:
- "The builder requested using id = smart" : that's the Build Chain, hopefully @Ginxo can help here
- "Jenkins" : that's the new pipeline, mistakenly configured to run on any branch; already fixed in the job configuration by allowing only the master branch

File: optaplanner-quickstarts/quarkus-school-timetabling/build.gradle
Comment (rsynek) [inline optaplanner-quickstarts/quarkus-school-timetabling/build.gradle:2]: [There](https://github.com/kiegroup/optaplanner/blob/2c1e0c1e729ba26df4242fdfd723563ba56805b8/optaplanner-quickstarts/quarkus-school-timetabling/build.gradle#L17) is also the universe-bom version, but I guess it's not yet available, is it?

Comment (ge0ffrey) [reply optaplanner-quickstarts/quarkus-school-timetabling/build.gradle:2]: for now, I am leaving using the universe bom for our quickstarts of the complexity :)
We can switch to it when we figure out a good way to be able to use it reliably...

**Comment by ge0ffrey** [comment]
Philosophical question: When is it ok to merge a PR? When all CI's are green or when none of the CI's result in a "true negative" (= AKA false negatives are ignored)?
Are contributions innocent or guilty until proven other wise? :)

**Comment by triceo** [comment]
> Are contributions innocent or guilty until proven otherwise? :)

I wouldn't look for a general rule. There will always be judgment calls.

Build chain 11 always fails.
But Jenkins, I think, really should pass.

**Comment by Ginxo** [comment]
> Ad CI failures:
> 
> * "The builder requested using id = smart" : that's the Build Chain, hopefully @Ginxo can help here
> * "Jenkins" : that's the new pipeline, mistakenly configured to run on any branch; already fixed in the job configuration by allowing only the master branch

@rsynek [takari libraries](http://takari.io/book/30-team-maven.html) were missing and they are needed by most of the projects in the build chain.
PR created, please review https://github.com/kiegroup/optaplanner/pull/963
once it's merged we should relaunch `Build Chain / Maven Build (8) (pull_request)`

**Pull request merged**
